services/workflows-service/src/business-report/business-report.repository.ts

```diff
@@ -34,4 +34,12 @@ export class BusinessReportRepository {
       this.scopeService.scopeFindMany(args, projectIds),
     );
   }
+  async upsert<T extends Prisma.BusinessReportUpsertArgs>(
+    args: Prisma.SelectSubset<T, Prisma.BusinessReportUpsertArgs>,
+    projectIds: TProjectIds,
+  ) {
+    args.where = { ...args.where, project: { id: { in: projectIds } } };
```

Comment:
Its a bit risky to have such a function, in case of the wrong where clause you can do some damage.

for that case add additional required 'id' field and do that upsert for specific row at once

Reply:
לא הבנתי, תעבור על זה שוב. אני לא כזה מבין איך זה יכול להיות מסוכן